DFC.ServiceTaxonomy.Events/Handlers/PublishToEventGridHandler.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Threading.Tasks;
+using DFC.ServiceTaxonomy.Events.Configuration;
+using DFC.ServiceTaxonomy.Events.Models;
+using DFC.ServiceTaxonomy.Events.Services.Interfaces;
+using DFC.ServiceTaxonomy.GraphSync.GraphSyncers.Interfaces;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Handlers;
+
+namespace DFC.ServiceTaxonomy.Events.Handlers
+{
+    public class PublishToEventGridHandler : ContentHandlerBase
+    {
+        private readonly IOptionsMonitor<EventGridConfiguration> _eventGridConfiguration;
+        private readonly IEventGridContentClient _eventGridContentClient;
+        private readonly IGraphSyncHelper _graphSyncHelper;
+        private readonly IPublishedContentItemVersion _publishedContentItemVersion;
+        private readonly IPreviewContentItemVersion _previewContentItemVersion;
+        private readonly ILogger<PublishToEventGridHandler> _logger;
+
+        private static readonly DeletedContentItemVersion _deletedContentItemVersion = new DeletedContentItemVersion();
+
+        public PublishToEventGridHandler(IOptionsMonitor<EventGridConfiguration> eventGridConfiguration, IEventGridContentClient eventGridContentClient, IGraphSyncHelper graphSyncHelper, IPublishedContentItemVersion publishedContentItemVersion, IPreviewContentItemVersion previewContentItemVersion, ILogger<PublishToEventGridHandler> logger)
+        {
+            _eventGridConfiguration = eventGridConfiguration;
+            _eventGridContentClient = eventGridContentClient;
+            _graphSyncHelper = graphSyncHelper;
+            _publishedContentItemVersion = publishedContentItemVersion;
+            _previewContentItemVersion = previewContentItemVersion;
+            _logger = logger;
+        }
+
+        public override async Task DraftSavedAsync(SaveDraftContentContext context)
+        {
+            _logger.LogInformation("PublishToEventGridHandler::Inside DraftSavedAsync");
```

Comment:
`PublishToEventGridHandler` will be logged automatically by the generic `ILogger`, so no need to also be part of the message. Also, if you want to log the method name, could you please use `nameof`.

Reply:
Ah forgot to remove these, they were just to help Simon and I trace what was going on when testing.